src/main/java/com/opencart/pages/HeaderPage.java

```diff
@@ -0,0 +1,104 @@
+package com.opencart.pages;
+
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+public class HeaderPage extends BasePage {
+
+    @FindBy(xpath = ".//*[@title='My Account']")
+    private WebElement myAccountButton;
+
+    @FindBy(xpath = ".//*[contains (@href,'register')]")
+    private WebElement registerButton;
+
+    @FindBy(xpath = ".//*[contains (@href, 'login')]")
+    private WebElement loginButton;
+
+    @FindBy(id = "form-currency")
+    private WebElement currencyButton;
+
+    @FindBy(name = "EUR")
+    private WebElement euroButton;
+
+    @FindBy(name = "USD")
+    private WebElement usdButton;
+
+    @FindBy(name = "GBP")
+    private WebElement gbpButton;
+
+    @FindBy(name = "UAH")
+    private WebElement uahButton;
+
+    @FindBy(css = "div.product-thumb.transition:first-child")
+    private WebElement macBookCost;
+
+    @FindBy(xpath = ".//*[contains (@href, 'account/logout')]")
+    private WebElement logoutButton;
+
+    @FindBy(css = "input[name = 'search']")
+    private WebElement searchInput;
+
+    @FindBy(css = "* button > i.fa.fa-search")
+    private WebElement searchButton;
+
+
+    public WebElement getMyAccountButton() {
+        wait.until(ExpectedConditions.visibilityOf(myAccountButton));
+        return myAccountButton;
+    }
+
+
+    public WebElement getLoginButton() {
+        wait.until(ExpectedConditions.visibilityOf(loginButton));
+        return loginButton;
+    }
+
+    public WebElement getCurrencyButton() {
+        wait.until(ExpectedConditions.visibilityOf(currencyButton));
+        return currencyButton;
+    }
+
+    public WebElement getRegisterButton() {
+        wait.until(ExpectedConditions.visibilityOf(registerButton));
+        return registerButton;
+    }
+
+    public WebElement getCurrencyButton(String currencyName) {
+        switch (currencyName.toLowerCase()) {
+            case "eur":
+                wait.until(ExpectedConditions.visibilityOf(euroButton));
+                return euroButton;
+            case "usd":
+                wait.until(ExpectedConditions.visibilityOf(usdButton));
+                return usdButton;
+            case "gbp":
+                wait.until(ExpectedConditions.visibilityOf(gbpButton));
+                return gbpButton;
+            case "uah":
+                wait.until(ExpectedConditions.visibilityOf(gbpButton));
+                return uahButton;
+            default: {
+                System.err.println("\nCan not switch to " + currencyName + " currency because this currency is not available. Please try again with another currency.");
+                return null;
+            }
+        }
+    }
+
+    public WebElement getMacBookCost() {
+        wait.until(ExpectedConditions.visibilityOf(macBookCost));
+        return macBookCost;
+    }
```

Comment:
I will delete the method "getMacBookCost" and locator for finding WebElements "macBookCost" because I don't use it now. I will use method verifyAllProductsPrices() to find and check the prices of all product Containers on the page (not just one product - MacBook). Screenshot is added.

I have also added "wait.until(ExpectedConditions.visibilityOfAllElements(productContainers));" to method getProductContainers() 
<img width="846" alt="Screen Shot 2020-11-19 at 12 42 26" src="https://user-images.githubusercontent.com/73054470/99655679-af3c6880-2a64-11eb-892b-cde7a6e9e1bf.png">
<img width="864" alt="Screen Shot 2020-11-19 at 12 44 13" src="https://user-images.githubusercontent.com/73054470/99655915-ee6ab980-2a64-11eb-9673-77b5bbae7c3b.png">